CMake/kwiver-setup-python.cmake

```diff
@@ -0,0 +1,118 @@
+###
+# Finds the python binaries, libraries, inclue, and site-packages paths
+
+
+set(KWIVER_PYTHON_VERSION "2" CACHE STRING "Python version to use: 3 or 2")
+set_property(CACHE KWIVER_PYTHON_VERSION PROPERTY STRINGS "3" "2")
+
+# If we change python versions re-find the bin, include, and libs
+if (NOT _prev_kwiver_python_version STREQUAL KWIVER_PYTHON_VERSION)
+  # but dont clobber initial settings in the instance they are specified via
+  # commandline (e.g  cmake -DPYTHON_EXECUTABLE=/my/special/python)
+  if (_prev_kwiver_python_version)
+    message(STATUS "The Python version changed; refinding the interpreter")
+    unset(PYTHON_EXECUTABLE CACHE)
+    unset(PYTHON_INCLUDE_DIR CACHE)
+    unset(PYTHON_LIBRARY CACHE)
+    unset(PYTHON_LIBRARY_DEBUG CACHE)
+  endif()
+endif()
+
+if (KWIVER_PYTHON_VERSION MATCHES "^3.*")
+  find_package(PythonInterp 3.4 REQUIRED)
+  find_package(PythonLibs 3.4 REQUIRED)
+else()
+  find_package(PythonInterp 2.7 REQUIRED)
+  find_package(PythonLibs 2.7 REQUIRED)
+endif()
+
+# Make a copy so we can determine if the user changes python versions
+set(_prev_kwiver_python_version "${KWIVER_PYTHON_VERSION}" CACHE INTERNAL
+  "allows us to determine if the user changes python version")
+
+
+###
+#
+# Get canonical directory for python site packages.
+# It varys from system to system.
+#
+execute_process(
+  COMMAND "${PYTHON_EXECUTABLE}" -c "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(prefix=''))"
+  RESULT_VARIABLE proc_success
+  OUTPUT_VARIABLE python_site_packages
+  )
+
+if(NOT ${proc_success} EQUAL 0)
+  message(FATAL_ERROR "Request for python site-packages location failed with error code: ${proc_success}")
+endif()
+
+# Current usage determines most of the path in alternate ways.
+# All we need to supply is the '*-packages' directory name.
+# Customers could be converted to accept a larger part of the path from this function.
+string( REGEX MATCH "dist-packages" result ${python_site_packages} )
+if (result)
+  set( python_sitename "dist-packages")
+else()
+  set( python_sitename "site-packages")
+endif()
+string(STRIP "${python_site_packages}" python_site_packages)
+message(STATUS "Python site-packages to install into: ${python_site_packages}")
+message(STATUS "python_site_packages = ${python_site_packages}")
+message(STATUS "python_sitename = ${python_sitename}")
+
+
+execute_process(
+  COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; print(sys.version[0:3])"
```

Comment:
We're looking for the version of python found while finding / requiring a specific version above?  Should this be checked against the requested version of python?

Reply:
We request the major version above, this grabs the major + minor version. We could add a check to make sure they agree, pending we don't rework how the versioning is done.